### DISCUSSION TITLE
container port not reachable in pod

### DISCUSSION THREAD

**Discussion by realSConway** [containers/podman#17328] [General]
### Issue Description

Hello experts,

I'm having issues connecting to listening port of container which is part of a pod.

```
cat /etc/os-release 
NAME="openSUSE MicroOS"
# VERSION="20230130"
ID="opensuse-microos"
ID_LIKE="suse opensuse opensuse-tumbleweed"
VERSION_ID="20230130"
PRETTY_NAME="openSUSE MicroOS"
ANSI_COLOR="0;32"
CPE_NAME="cpe:/o:opensuse:microos:20230130"
BUG_REPORT_URL="https://bugs.opensuse.org"
HOME_URL="https://www.opensuse.org/"
DOCUMENTATION_URL="https://en.opensuse.org/Portal:MicroOS"
LOGO="distributor-logo-MicroOS"

$ podman version
Client:       Podman Engine
Version:      4.3.1
API Version:  4.3.1
Go Version:   go1.17.13
Built:        Tue Jan 17 01:00:00 2023
OS/Arch:      linux/amd64
```

### Steps to reproduce the issue

Create a pod:
`podman pod create --name node --publish 127.0.0.1:50002:50002`
Then create container in pod:
`podman create --name ers --pod service localhost/project`
start service
`podman start service`
I check the status of the service and see it it up:
`e3dea3659a11  localhost/project:latest                 bash         5 hours ago   Up 5 hours ago             127.0.0.1:50002->50002/tcp  service`

This service container runs a service that listens on port 50002.

However when I nc to server:port, I get error: `nc: connect to 127.0.0.1 port 50002 (tcp) failed: Connection refused`

Run netshoot container in pod:
`podman run --name netshoot --pod bitcoin nicolaka/netshoot`

With netshoot I have access to more networking tools.
here are the results:
```
$ ss -tulp                  
Netid    State    Recv-Q    Send-Q         Local Address:Port         Peer Address:Port    Process    

 f43911b97669  ~  netstat -tulpn | grep LISTEN

$ ss -tulp
Netid    State    Recv-Q    Send-Q         Local Address:Port         Peer Address:Port    Process    

$ nc -zv 127.0.0.1 50002
nc: connect to 127.0.0.1 port 50002 (tcp) failed: Connection refused
```

### Describe the results you received

Describe the results you received

### Describe the results you expected

Describe the results you expected

### podman info output

```shell
If you are unable to run podman info for any reason, please provide the podman version, operating system and its version and the architecture you are running.
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by vrothberg** [comment]
Thanks for reaching out, @realSConway.  I converted it to a discussion as it looks more like a question rather than a bug to me.

Could it be a bug in your application `localhost/project`?  I suggest trying it with a known-to-work application (e.g., nginx).

**Reply by realSConway** [reply]
Hi Vrothberg,
Thanks for getting back to me and moving this to discussion.

I found the problem, was running pod with `127.0.0.1:50002:50002`, it should be without binding to localhost (127.0.0.1).

**Reply by vrothberg** [reply]
Glad you could get it running, thank you!
